src/shared/ethtool-util.c

```diff
@@ -434,39 +435,42 @@ int ethtool_set_nic_buffer_size(int *ethtool_fd, const char *ifname, const netde
         return 0;
 }
 
-static int get_stringset(int ethtool_fd, struct ifreq *ifr, int stringset_id, struct ethtool_gstrings **ret) {
+static int get_stringset(int ethtool_fd, const char *ifname, enum ethtool_stringset stringset_id, struct ethtool_gstrings **ret) {
         _cleanup_free_ struct ethtool_gstrings *strings = NULL;
         struct {
                 struct ethtool_sset_info info;
                 uint32_t space;
         } buffer = {
-                .info = {
-                        .cmd = ETHTOOL_GSSET_INFO,
-                        .sset_mask = UINT64_C(1) << stringset_id,
-                },
+                .info.cmd = ETHTOOL_GSSET_INFO,
+                .info.sset_mask = UINT64_C(1) << stringset_id,
+        };
+        struct ifreq ifr = {
+                .ifr_data = (void*) &buffer,
         };
-        unsigned len;
+        uint32_t len;
         int r;
 
         assert(ethtool_fd >= 0);
-        assert(ifr);
+        assert(ifname);
         assert(ret);
 
-        ifr->ifr_data = (void *) &buffer.info;
+        strscpy(ifr.ifr_name, IFNAMSIZ, ifname);
 
-        r = ioctl(ethtool_fd, SIOCETHTOOL, ifr);
+        r = ioctl(ethtool_fd, SIOCETHTOOL, &ifr);
```

Comment:
maybe write as:

```c
r = ioctl(ethtool_fd, SIOCETHTOOL, &(struct ifreq) { .ifr_data = &buffer.info });
```

Given that you don't actually care about the ifreq after the ioctl, but just the `buffer` a compound literal sounds good to me.

but really doesn't matter

Reply:
No, `ifr.ifr_name` also needs to be set.